src/main/java/com/purbon/kafka/topology/serdes/TopicCustomDeserializer.java

```diff
@@ -86,17 +86,28 @@ public TopicImpl deserialize(JsonParser parser, DeserializationContext context)
     TopicImpl topic =
         new TopicImpl(name, producers, consumers, optionalDataType, config, this.config);
 
+    // Amends from issue 155
+    List<TopicSchemas> schemas = new ArrayList<>();
     Optional.ofNullable(rootNode.get("schemas"))
         .ifPresent(
-            node -> {
-              topic.setSchemas(
-                  Optional.of(
-                      new TopicSchemas(
-                          Optional.ofNullable(node.get("key.schema.file")),
-                          Optional.ofNullable(node.get("value.schema.file")))));
-            });
+            node ->
+                node.elements()
```

Comment:
The problem you are seeing here is when parsing the format with only one elements, this is not an array. For this you need to change this lines to be either returning node.elements or building an iterator when the node is a JsoNode.

Try something like this around these lines:

```java
 Iterator<JsonNode> elements =
                  node instanceof ArrayNode ? node.elements() : Arrays.asList(node).iterator();
              elements.forEachRemaining(
                  node1 -> {
                    TopicSchemas schema =
                        new TopicSchemas(
                            Optional.ofNullable(node1.get("key.schema.file")),
                            Optional.ofNullable(node1.get("value.schema.file")));
                    schemas.add(schema);
```